apps/dcos_dns/src/dcos_dns_router.erl

```diff
@@ -12,15 +12,22 @@
 %% @doc Resolvers based on a set of "questions"
 -spec(upstreams_from_questions(dns:questions()) ->
     {[upstream()] | internal, binary()}).
-upstreams_from_questions([#dns_query{name=Name}]) ->
-    Labels = dcos_dns_app:parse_upstream_name(Name),
-    find_upstream(Labels);
-upstreams_from_questions([Question|Others]) ->
-    %% There is more than one question. This is beyond our capabilities at the moment
-    dcos_dns_metrics:update([dcos_dns, ignored_questions], length(Others), ?COUNTER),
-    Result = upstreams_from_questions([Question]),
-    lager:debug("~p will be forwarded to ~p", [Others, Result]),
-    Result.
+upstreams_from_questions(Questions) ->
+    UpstreamAnswers = fun (#dns_query{name=Name}) ->
+        Labels = dcos_dns_app:parse_upstream_name(Name),
+        find_upstream(Labels)
+    end,
+    Answers = lists:map(UpstreamAnswers, Questions),
+
+    case lists:usort(Answers) of
+        [Answer] ->
+            Answer;
+        MixedAnswers ->
+            InternalResolver = {internal, <<".">>},
+            lager:notice("DNS question ~p has mixed upstream ~p. forwarding ~p",
```

Comment:
`notice` is used when we are changing some state, please use `warning` or `error in such cases.